src/Catalog/CatalogWriterBase.cs

```diff
@@ -70,7 +70,7 @@ public Task Commit( IGraph commitMetadata, CancellationToken cancellationToken)
 
             if (_batch.Count == 0)
             {
-                return;
+                return null;
```

Comment:
Q: we could return `Enumerable.Empty<string>()` and avoid null-checks in caller code?